src/artemisa/component.js

```diff
@@ -1,23 +1,15 @@
 import React from 'react'
 import { connect } from 'react-redux'
-
 import { State } from '../core/model'
-import { isReceive } from '../core/actions'
-import { hasValueInKey } from '../util/object'
-import { shouldFetch } from '../core/reducer'
 import { isFunction, trueFn, identity } from '../util/function'
-export const ARTEMISA = 'ARTEMISA'
-
-export const isArtemisaType = type => type && !!type.match(/^ARTEMISA/)
-export const isArtemisaReceive = action => isReceive(action) && isArtemisaType(action.originType)
-export const isArtemisaAction = action => isArtemisaType(action.type)
-
-export function storagePropertyNameForAction(action) {
-  return action.originType.slice(ARTEMISA.length + 1)
-}
```

Comment:
I've moved all the "action dispatch" related logic to a new file `dispatch.js` so that the component remains clean and short, and delegates all actions related functions to that one